telegram/ext/filters.py

```diff
@@ -865,73 +865,83 @@ class user(BaseFilter):
         Args:
             user_id(:obj:`int` | List[:obj:`int`], optional): Which user ID(s) to allow
                 through.
             username(:obj:`str` | List[:obj:`str`], optional): Which username(s) to allow
                 through. If username starts with '@' symbol, it will be ignored.
+            allow_empty(:obj:`bool`, optional): Whether updates should be processed, if no user
+                is specified in :attr:`user_ids` and :attr:`usernames`. Defaults to :obj:`False`
 
         Raises:
             ValueError: If chat_id and username are both present, or neither is.
 
         """
 
-        def __init__(self, user_id=None, username=None):
-            if (user_id is None) == (username is None):
-                raise ValueError('One and only one of user_id or username must be used')
+        def __init__(self, user_id=None, username=None, allow_empty=False):
+            self.allow_empty = allow_empty
+            self.__lock = Lock()
 
-            self._user_ids_lock = Lock()
-            self._usernames_lock = Lock()
+            self._user_ids = {}
+            self._usernames = {}
 
-            # Initialize in a way that will not fail the first setter calls
-            self._user_ids = user_id
-            self._usernames = username
-            # Actually initialize
             self.user_ids = user_id
             self.usernames = username
 
+        def _set_user_ids(self, user_id):
+            if user_id is None:
+                self._user_ids = {}
+            elif isinstance(user_id, int):
+                self._user_ids = {user_id}
+            else:
+                self._user_ids = set(user_id)
+
+        def _set_usernames(self, username):
+            if username is None:
+                self._usernames = {}
+            elif isinstance(username, str):
+                self._usernames = {username.replace('@', '')}
+            else:
+                self._usernames = set([user.replace('@', '') for user in username])
+
         @property
         def user_ids(self):
-            with self._user_ids_lock:
-                return self._user_ids
+            with self.__lock:
+                return frozenset(self._user_ids) if self._user_ids is not None else None
 
         @user_ids.setter
         def user_ids(self, user_id):
-            if (user_id is None) == (self.usernames is None):
-                raise ValueError('One and only one of user_id or username must be used')
-            with self._user_ids_lock:
-                if user_id is None:
-                    self._user_ids = None
-                elif isinstance(user_id, int):
-                    self._user_ids = set([user_id])
-                else:
-                    self._user_ids = set(user_id)
+            with self.__lock:
+                if user_id and self._usernames:
+                    raise RuntimeError(('Can\'t set user_id in conjunction with (already set) '
+                                        'usernames.'))
```

Comment:
Why the extra `(`? it won't be treated as a tuple (as you didn't add a comma at the end)

Also, being the nitpicker I am, it really hurts my eyes to see backspace: `Can\'t` 
This is a legitimate place to use double quotes so you won't have to escape the `'`.

Reply:
For some reason I thought without the extra `(` that would be invalid syntax … removing.